paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java

```diff
@@ -125,14 +133,16 @@ List<Map<String, String>> expire(LocalDateTime now, long commitIdentifier) {
 
     private List<Map<String, String>> doExpire(
             LocalDateTime expireDateTime, long commitIdentifier) {
-        List<Map<String, String>> expired = new ArrayList<>();
+        List<List<String>> partValues = new ArrayList<>();
         for (PartitionEntry partition : strategy.selectExpiredPartitions(scan, expireDateTime)) {
             Object[] array = strategy.convertPartition(partition.partition());
-            Map<String, String> partString = strategy.toPartitionString(array);
-            expired.add(partString);
-            LOG.info("Expire Partition: {}", partString);
+            partValues.add(strategy.toPartitionValue(array));
         }
-        if (!expired.isEmpty()) {
+
```

Comment:
Use follow code could init partValues with capacity.

List<PartitionEntry> partitionEntries = strategy.selectExpiredPartitions(scan, expireDateTime);
        List<List<String>> partValues = new ArrayList<>(partitionEntries.size());
        for (PartitionEntry partition : partitionEntries) {
            Object[] array = strategy.convertPartition(partition.partition());
            partValues.add(strategy.toPartitionValue(array));
        }